applications/adv_networking_bench/cpp/dpdk_bench_op_rx.h

```diff
@@ -136,91 +139,157 @@ class AdvNetworkingBenchDefaultRxOp : public Operator {
+      // Free buffers for packets which have already been aggregated to the GPU again, in case some
+      // of it got completed since the beginning of `compute`, so we have extra space in `batch_q_`.
+      // NOTE: In CPU-only mode we can free earlier (after memcopy), but to keep it simple we free
       //       at the same point as we do in GPUDirect mode
       free_processed_packets();
       if (batch_q_.size() == num_concurrent) {
+        // Not enough buffers available to process the packets, drop the packets from this burst
         HOLOSCAN_LOG_ERROR("Fell behind in processing/copying on GPU!");
         adv_net_free_all_pkts_and_burst(burst);
         return;
       }
 
+      // Process/aggregate
       if (gpu_direct_.get()) {
+        // GPUDirect mode: we copy the payload (referenced in h_dev_ptrs_)
+        // to a contiguous memory buffer (full_batch_data_d_)
+        // NOTE: there is no actual reordering since we use the same order as packets came in,
+        //   but they would be reordered if h_dev_ptrs_ was filled based on packet sequence id.
         simple_packet_reorder(static_cast<uint8_t*>(full_batch_data_d_[cur_batch_idx_]),
                               h_dev_ptrs_[cur_batch_idx_],
                               nom_payload_size_,
                               batch_size_.get(),
                               streams_[cur_batch_idx_]);
 
       } else {
+          // Non GPUDirect mode: we copy the payload on host-pinned memory (in full_batch_data_h_)
+          // to a contiguous memory buffer on the GPU (full_batch_data_d_)
+          // NOTE: there is no reordering support here at all
           cudaMemcpyAsync(full_batch_data_d_[cur_batch_idx_],
                           full_batch_data_h_[cur_batch_idx_],
                           batch_size_.get() * nom_payload_size_,
                           cudaMemcpyDefault,
                           streams_[cur_batch_idx_]);
       }
 
+      // Keep track of the CUDA work (reorder kernel or copy from CPU) so we can free the packets
+      // for its bursts when the work is done.
       cudaEventRecord(events_[cur_batch_idx_], streams_[cur_batch_idx_]);
-
       cur_batch_.evt = events_[cur_batch_idx_];
       batch_q_.push(cur_batch_);
-      cur_batch_.num_bursts = 0;
 
+      // CUDA Error checking
       if (cudaGetLastError() != cudaSuccess) {
         HOLOSCAN_LOG_ERROR("CUDA error with {} packets in batch and {} bytes total",
                             batch_size_.get(),
                             batch_size_.get() * nom_payload_size_);
         exit(1);
       }
 
+      // Update structs for the next batch
+      cur_batch_.num_bursts = 0;
       cur_batch_idx_ = (++cur_batch_idx_ % num_concurrent);
+
+      // NOTE: output for the next operator is full_batch_data_d_, once the CUDA event is completed
     }
   }
 
  private:
+  // TODO: make configurable?
```

Comment:
Are ANO tasks tracked in Jira? Could we add TODOs?

Reply:
At this stage it's still hand-wavy, it's more to keep track of notes alongside the code instead of in a google docs. Normalizing our process for taking ANO to production support is next and will depend on stakeholders feedback following my presentation.